### PR TITLE
Set minimum size for empty matplotlib circuit drawings

### LINKED ISSUE #6101: drawing the schedule of an empty circuit raises warning from matplotlib
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.8
- **Operating system**: MacOS

### What is the current behavior?

```
from qiskit import schedule, QuantumCircuit, transpile
from qiskit.test.mock import FakeOpenPulse2Q

circuit = QuantumCircuit(1)

schedule(circuit, FakeOpenPulse2Q()).draw()
```
```
.../qiskit-terra/qiskit/visualization/pulse_v2/plotters/matplotlib.py:160: UserWarning: Attempting to set identical left == right == 0.0 results in singular transformations; automatically expanding.
  self.ax.set_xlim(*axis_config.window)
.../qiskit-terra/venv/lib/python3.8/site-packages/matplotlib/tight_bbox.py:71: RuntimeWarning: divide by zero encountered in double_scalars
  fig.patch.set_bounds(x0 / w1, y0 / h1,
.../qiskit-terra/venv/lib/python3.8/site-packages/matplotlib/tight_bbox.py:72: RuntimeWarning: divide by zero encountered in double_scalars
  fig.bbox.width / w1, fig.bbox.height / h1)
.../qiskit-terra/venv/lib/python3.8/site-packages/matplotlib/patches.py:769: RuntimeWarning: invalid value encountered in double_scalars
  y1 = self.convert_yunits(self._y0 + self._height)
.../venv/lib/python3.8/site-packages/matplotlib/transforms.py:2013: RuntimeWarning: invalid value encountered in double_scalars
  self._mtx[1, 2] += ty
```

### What is the expected behavior?

I think the actual output drawing is correct. 
image

However, I dont think these warnings are useful and they should be removed.

### Suggested solutions

I think giving some dummy values is probably the cleanest way to remove these obscure warnings.

### PR DESCRIPTION
**Pull request by scapape**
### Summary

Fixes #6101 

### Details and comments

Empty circuits drawn using schedule(circuit, FakeOpenPulse2Q()).draw() raised warnings from matplotlib because figure dimensions were 0 (axis_config.window = (0, 0); canvas_height = 0). This PR sets dummy values to:
1) Avoid the warnings from matplotlib;
2) Draw an empty circuit consistent with a not-empty circuit (see image below).
image

The following conditions have been added.
1 (to set a minimum figure height):

        canvas_height = max(canvas_height, 0.1)

And 2 (to place figure title to the left similar to non-empty circuit):

        if axis_config.window == (0, 0):
            self.ax.set_xlim(0, 1)
        else:
            self.ax.set_xlim(*axis_config.window)

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
Thanks for looking at this, and impressive speed between being assigned and making the PR!  I left one question about what you'd intended, but this looks good to me and solves the problem with no damaging side effects I can think of.

File: qiskit/visualization/pulse_v2/plotters/matplotlib.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def __init__(self, canvas: core.DrawerCanvas, axis: Optional[plt.Axes] = None):
 
         if axis is None:
             fig_h = canvas_height * self.canvas.formatter["general.fig_chart_height"]
+            if fig_h == 0:
+                fig_h = 0.1
```
[the remark this answers is not in the document]
Comment (scapape) [reply qiskit/visualization/pulse_v2/plotters/matplotlib.py:57]: Thanks, @jakelishman. Yes, I can look into that ie setting `canvas_height` to min size

[the remark this answers is not in the document]
Comment (scapape) [reply qiskit/visualization/pulse_v2/plotters/matplotlib.py:57]: Following your suggestion, setting `canvas_height` minimum size replaces setting `fig_h` minimum size:
```
        min_canvas_height = 0.1
        if canvas_height < min_canvas_height:
            canvas_height = min_canvas_height
```

**Review by jakelishman — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**